.github/workflows/tests.yaml

```diff
@@ -1,23 +1,32 @@
 ---
 name: tests
 
-on:
-  push:
-    branches-ignore:
-    - 'development/**'
+on: push
+
+permissions:
+  packages: write
+  id-token: write
+  contents: read
 
 jobs:
   test:
     runs-on: ubuntu-latest
     steps:
     - name: Checkout
-      uses: actions/checkout@v2
-    - uses: actions/setup-node@v2
+      uses: actions/checkout@v4
+    - uses: actions/setup-node@v4
       with:
         node-version: '16'
+        registry-url: https://npm.pkg.github.com
     - name: install dependencies
       run: yarn install --frozen-lockfile
     - name: lint markdown
       run: yarn lint_md
     - name: lint javascript
       run: yarn lint
+    - name: Set development version
+      run: yarn version --no-git-tag-version --new-version "0.0.0-sha.${{ github.sha }}"
```

Comment:
As discussed with @BourgoisMickael:

- Publish on dev branch with the version number instead of 0.0.0-sha
- And use a short sha (7).

Reply:
@BourgoisMickael it's now done, as requested.